src/tools/components/SequenceSearchLoader.tsx

```diff
@@ -0,0 +1,245 @@
+        header: '',
+        sequence: '',
+        valid: true,
+        likelyType: null,
+        message: null,
+        // name as a NCBI ID formatted UniProt-style
+        name: entry.uniProtkbId
+          ? `${entry.entryType === EntryType.REVIEWED ? 'sp' : 'tr'}|${
+              entry.primaryAccession
+            }|${entry.uniProtkbId}`
+          : '',
+      },
+    ]);
+  }, [entry, onLoad, urlForAccessionOrID, accessionOrID]);
+
+  const handlePaste = useCallback(
+    async (event: ClipboardEvent) => {
+      const pastedContent = event.clipboardData.getData('text/plain');
+      // use a Set to remove duplicate
+      const potentialAccessions = new Set(
+        pastedContent.split(/\W+/).map((text) => text.toUpperCase())
+      );
+      if (!potentialAccessions.size) {
+        return;
+      }
+
+      // prevent displaying the content of the clipboard and empty the input
+      event.preventDefault();
+      setAccessionOrID('');
+      setPasteLoading(true);
+
+      try {
+        const parsedSequences = [];
+        const errors = [];
+        for (const acc of potentialAccessions) {
+          try {
+            const url = getURLForAccessionOrID(acc);
+            if (!url) {
+              continue; // eslint-disable-line no-continue
+            }
+            // eslint-disable-next-line no-await-in-loop
+            let { data } = await fetchData<NetworkResponses>(url);
+            if ('results' in data) {
+              [data] = data.results;
+            }
+
+            parsedSequences.push({
+              raw: `${entryToFASTAWithHeaders(data)}\n`,
+              // no need to fill the rest, it will be parsed again later
+              header: '',
+              sequence: '',
+              valid: true,
+              likelyType: null,
+              message: null,
+              // name as a NCBI ID formatted UniProt-style
+              name: data.uniProtkbId
+                ? `${data.entryType === EntryType.REVIEWED ? 'sp' : 'tr'}|${
+                    data.primaryAccession
+                  }|${data.uniProtkbId}`
+                : '',
+            });
+          } catch (_) {
+            errors.push(acc);
+          }
+        }
+        onLoad(parsedSequences);
+        if (errors.length) {
+          dispatch(
+            addMessage({
+              id: v1(),
+              content: `There was an issue retrieving sequence data for: ${errors.join(
+                ', '
+              )}`,
+              format: MessageFormat.POP_UP,
+              level: MessageLevel.FAILURE,
+            })
+          );
+        }
+      } catch (error) {
+        /**/
+      } finally {
+        setPasteLoading(false);
+      }
+    },
+    [onLoad, dispatch]
+  );
+
+  return (
+    <SearchInput
+      isLoading={loading || pasteLoading}
+      onChange={(e: ChangeEvent<HTMLInputElement>) =>
+        setAccessionOrID(e.target.value)
+      }
+      onPaste={handlePaste}
```

Comment:
As I mentioned before, I think it's good if we don't use clipboard events, since they aren't stable yet.